src/fipp/ednize.clj

```diff
@@ -91,3 +80,31 @@
 
 (defn record->tagged [x]
   (tagged-literal (-> x class .getName symbol) (into {} x)))
+
+(def java-sql-timestamp
+  (try
+    (Class/forName "java.sql.Timestamp")
+    (catch ClassNotFoundException _
+      false)))
+
+(def java-sql-timestamp?
+  (and java-sql-timestamp
+       (try
+         ;; strangely, this can fail even if the previous statement succeeded:
+         (require '[clojure.instant])
+         true
+         (catch ExceptionInInitializerError _
+           false))))
+
+(when java-sql-timestamp?
+  (require '[clojure.instant])
+  (eval `(extend-protocol IEdn
+           ~(-> java-sql-timestamp .getName symbol)
```

Comment:
Is there really no cleaner approach to avoiding mentioning a class name than eval/syntax-quote/unquote? I'd imagine one could use the `extend` function instead of the `extend-protocol` macro to clean this up a bit.